core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -89,14 +99,33 @@ class KafkaRequestHandlerPool(val brokerId: Int,
                               time: Time,
                               numThreads: Int) extends Logging with KafkaMetricsGroup {
 
+  private val threadPoolSize: AtomicInteger = new AtomicInteger(numThreads)
   /* a meter to track the average free capacity of the request handlers */
   private val aggregateIdleMeter = newMeter("RequestHandlerAvgIdlePercent", "percent", TimeUnit.NANOSECONDS)
 
   this.logIdent = "[Kafka Request Handler on Broker " + brokerId + "], "
-  val runnables = new Array[KafkaRequestHandler](numThreads)
+  val runnables = new mutable.ArrayBuffer[KafkaRequestHandler](numThreads)
```

Comment:
Do we need any locking for this?

Reply:
Thought it didn't need since we only do one update at a time, but added anyway to be safe.

Reply:
Yeah, I guess `DynamicBrokerConfig` only handles one change at a time, but maybe better to be safe.